sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/aio/hybrid_search_aggregator.py

```diff
@@ -147,21 +147,44 @@ async def _run_hybrid_search(self):
             self._format_final_results(drained_results)
             return
 
+        # Get the Components weights if any
+        if self._hybrid_search_query_info.get('componentWeights'):
+            component_weights = self._hybrid_search_query_info['componentWeights']
+            component_count = len(self._hybrid_search_query_info['componentQueryInfos'])
+            # If the number of weights is less than the number of components, we fill the rest with 1.0
+            # Unlikely to be the case since a bad request should be returned if weights are less than components,
+            # and any extra weights relative to components are ignored, but we handle it here anyway.
+            if len(component_weights) != component_count:
+                if len(component_weights) < component_count:
+                    # Fill the rest with 1.0
+                    component_weights = [component_weights[x] if x < len(component_weights)
+                                         else 1.0 for x in range(component_count)]
+                else:
+                    # Drop the excess values
+                    component_weights = component_weights[:component_count]
+        else:
+            # If no weights are provided, we default to 1.0 for all components
+            component_weights = [1.0] * len(self._hybrid_search_query_info['componentQueryInfos'])
+
         # Sort drained results by _rid
         drained_results.sort(key=lambda x: x['_rid'])
 
         # Compose component scores matrix, where each tuple is (score, index)
         component_scores = _retrieve_component_scores(drained_results)
 
-        # Sort by scores in descending order
-        for score_tuples in component_scores:
-            score_tuples.sort(key=lambda x: x[0], reverse=True)
+        # Sort by scores using component weights
+        for index, score_tuples in enumerate(component_scores):
+            # Negative Weights will change sorting from Descending to Ascending
+            ordering = self._hybrid_search_query_info['componentQueryInfos'][index]['orderBy'][0]
+            comparison_factor = not ordering.lower() == 'ascending'
+            #  pylint: disable=cell-var-from-loop
+            score_tuples.sort(key=lambda x: component_weights[index] * x[0], reverse=comparison_factor)
```

Comment:
We don't have to multiply the component weight here. The query plan will always report positive weights. The sort order would be based on the sign of the weight.

This code needs to be updated to

score_tuples.sort(key=lambda x: x[0], reverse=comparison_factor)

Reply:
The change has been made.